lib/Util.ts

```diff
@@ -18,8 +19,11 @@ export function getUrl(requestUrl: string, baseUrl?: string): string  {
     combined.protocol = combined.protocol || base.protocol;
     combined.auth = combined.auth || base.auth;
     combined.host = combined.host || base.host;
-    // path from requestUrl always wins
 
+    combined.pathname = path.join(base.pathname, combined.pathname);
```

Comment:
Instead of path.join, let's try path.resolve.
So 
'http://httpbin.org' and 'foo/bar' = 'http://httpbin.org/foo/bar'
'http://httpbin.org' and '/foo/bar' = 'http://httpbin.org/foo/bar'
'http://httpbin.org/baz' and 'foo/bar' = 'http://httpbin.org/baz/foo/bar'
'http://httpbin.org/baz' and '/foo/bar' = 'http://httpbin.org/foo/bar'

I sketched it out in a sketch PR. 

https://github.com/Microsoft/typed-rest-client/pull/28

See my tests.  You can merge in with your PR.  

Also note I bumped to 1.0 and will release with tag of preview.  After thinking about it, I think that's the best thing to do.

Thoughts?

Reply:
The problem with using `path.resolve` on a Windows system, is that an absolute path will include a drive letter. For the test case

`let res: string = util.getUrl('get/foo', 'http://httpbin.org/bar');`
`res` comes out to `http://httpbin.org/C:\bar\get\foo`